### DISCUSSION TITLE
Validate JWT token in Keycloak server without login page

### DISCUSSION THREAD

**Discussion by rcbandit111** [keycloak/keycloak#15881] [Archive]
I want to use Keycloak server with Spring Gateway. In my case I have a API to API communication (Server to Server) communication which I would like to protect using JWT token.

The question is how I can use Keycloak Authorization server to generate and validate JWT token having in mind that I don't have a login page. I need to:

1. Generate JWT using Keycloak API because I want to generate the JWT token from a customer portal page and put some custom data into JWT token payload.

2. When API request is send to Spring Gateway I want to validate the JWT token into Keycloak server.

Is this functionality possible with Spring gateway and Keycloak server?

**Comment by sschu** [comment]
Obtaining these tokens is possible using the client credentials grant (see https://datatracker.ietf.org/doc/html/rfc6749#section-4-4). In Keycloak this feature is called "service account", see https://www.keycloak.org/docs/latest/server_admin/#_service_accounts.
